magit-todos.el

```diff
@@ -397,15 +397,16 @@ Only necessary when option `magit-todos-update' is nil."
   "Show current item.
 If PEEK is non-nil, keep focus in status buffer window."
   (interactive)
-  (let* ((status-window (selected-window))
-         (item (oref (magit-current-section) value))
-         (buffer (magit-todos--item-buffer item)))
-    (pop-to-buffer buffer)
-    (magit-todos--goto-item item)
-    (when (derived-mode-p 'org-mode)
-      (org-show-entry))
-    (when peek
-      (select-window status-window))))
+  (if-let* ((status-window (selected-window))
+            (item (oref (magit-current-section) value))
+            (is-valid-item (cl-struct-p item))
```

Comment:
Is this necessary?  I'm guessing that if it's not a valid item, `item` will be nil, so the `if-let*` form will short-circuit there.

Reply:
It is necessary, because `item` is not `nil` even the item is not a valid item, e.g. when current section is on headings and `item` will be a string or a list and we need to test if the item is a valid structure.